enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -1466,6 +1511,8 @@ class AdjointGenerator
 
     Type *addingType = BO.getType();
 
+    setDiffe(&BO, Constant::getNullValue(BO.getType()), Builder2);
```

Comment:
Why do we need this? If this op is constant we should never set it, and if it is active we should always set it to the computed derivative?

Reply:
Because I am using `addToDiffe` I need to "reset" variables that are used in loops. If I stop using `addToDiffe` I can get rid of this.

Reply:
We should just use setDiffe for forward mode (and arguably can assert in addToDiffe if in forwardMode).

Reply:
Already working on this 😄

Reply:
Bump at potentially removing this?

Reply:
completely forgot to remove this 🙈